### PR TITLE
POL-237 Provide version info and print at start.

### PR DESCRIPTION
**Pull request by pilhuhn**
> ```
> 2020-04-21 21:57:01,346 INFO  [org.fly.cor.int.com.DbMigrate] (main) Current version of schema "public": 7
> 2020-04-21 21:57:01,349 INFO  [org.fly.cor.int.com.DbMigrate] (main) Schema "public" is up to date. No migration necessary.
> 2020-04-21 21:57:01,825 INFO  [Policies UI-Backend] (main) 
>     Build-date Tue Apr 21 21:51:07 CEST 2020 
>     on host [snert]
>     from branch [show_version]
>     with id 56e118f3c95e24cb012348acf385ddf239859dee
> 2020-04-21 21:57:01,945 INFO  [io.quarkus] (main) app 1.0-SNAPSHOT (powered by Quarkus 1.3.2.Final) started in 5.455s. Listening on: http://0.0.0.0:8080
> 2020-04-21 21:57:01,948 INFO  [io.quarkus] (main) Profile dev activated. Live Coding activated.
> 2020-04-21 21:57:01,948 INFO  [io.quarkus] (main) Installed features: [agroal, cache, cdi, flyway, hibernate-orm, hibernate-orm-panache, hibernate-validator, jaeger, jdbc-postgresql, mutiny, narayana-jta, rest-client, resteasy, resteasy-jsonb, smallrye-health, sm
> ```
> 

Signed-off-by: Heiko W. Rupp <hrupp@redhat.com>

### CODE REVIEW

**Comment by josejulio** [comment]
Travis is not happy about this.
```
[ERROR] Command execution failed.
java.io.IOException: Cannot run program "/home/travis/build/RedHatInsights/policies-ui-backend/scripts/calculate-version.sh" (in directory "/home/travis/build/RedHatInsights/policies-ui-backend"): error=2, No such file or directory
```

File: pom.xml [hunk 1]
```diff
@@ -146,6 +147,23 @@
...
+        <version>${exec-maven-plugin.version}</version>
+        <executions>
+          <execution><!-- Run our version calculation script -->
+            <id>Provide-git-data</id>
+            <phase>generate-sources</phase>
+            <goals>
+              <goal>exec</goal>
+            </goals>
+            <configuration>
+              <executable>${basedir}/scripts/calculate-version.sh</executable>
```
Comment (josejulio) [inline pom.xml:162]:
```suggestion
              <executable>${basedir}/helpers/calculate-version.sh</executable>
``` 
?

**Comment by pilhuhn** [comment]
see also https://travis-ci.org/github/RedHatInsights/policies-ui-backend/builds/677865830#L359

File: src/main/java/com/redhat/cloud/policies/app/JaxRsApplication.java [hunk 2]
```diff
@@ -32,12 +34,25 @@
...
-  // Produce access log
+  // Server init is done here, so we can do some more initialisation
+
   void observeRouter(@Observes Router router) {
+    //Produce access log
     Handler<RoutingContext> handler = new JsonAccessLoggerHandler(filterHealth);
     router.route().order(-1000).handler(handler);
+
+    // Produce build-info and log on startup
+    String info = String.format("\n    Build-date %s \n    on host [%s]\n    from branch [%s]\n    with id %s",
```
Comment (josejulio) [inline src/main/java/com/redhat/cloud/policies/app/JaxRsApplication.java:50]: Should we use `SHA` instead of `id` ? or maybe `commit id`.

**Comment by pilhuhn** [comment]
See https://travis-ci.org/github/RedHatInsights/policies-ui-backend/builds/678038076#L359

**Comment by burmanm** [comment]
This is unnecessary though. We already have that information in our containers:

sh-4.4$ env | grep OPENSH
OPENSHIFT_BUILD_SOURCE=https://github.com/RedHatInsights/policies-engine
OPENSHIFT_BUILD_NAME=policies-engine-29
AB_JOLOKIA_AUTH_OPENSHIFT=true
OPENSHIFT_BUILD_COMMIT=fc5190d330c93febf6aba7b19df6374f1180aebc
OPENSHIFT_BUILD_NAMESPACE=buildfactory
OPENSHIFT_BUILD_REFERENCE=master
sh-4.4$

s2i adds this information. So we only need to read: OPENSHIFT_BUILD_COMMIT env variable and print it.

**Comment by theute** [comment]
> s2i adds this information. So we only need to read: OPENSHIFT_BUILD_COMMIT env variable and print it.

Oh that's nice !

**Comment by pilhuhn** [comment]
@burmanm That's nice. How do we want to deal if not built on OpenShift s2i?

**Comment by burmanm** [comment]
Are we going to build without s2i at some point?

**Comment by pilhuhn** [comment]
There are no plans that I am aware of. I was more thinking about local builds or Travis builds or so.

**Pull request merged**